Add modlog monitoring

chris-y opened the pull request:
Add monitoring of modlogs.
Currently only supports ModRemovePosts (see #6)

Note this is currently collecting modlogs from *subscribed* communities, not *moderated* communities.  This is due to a bug in Lemmy where asking for Moderated communities appears to return All.

Also adds a facility to PM the user when their post gets removed, with the reason.

The deployment is set to not collect modlogs (function not active)
Suggest to run this on the modbots, but we need to sub to the correct communities with them before enabling.
Suggest to enable MODLOG_PM only after it has run at least once, otherwise we risk PMing somebody with an old report.

chris-y committed:
Add modlog monitoring for removed_posts
Add an option to PM the user when their post gets modded
Note this is currently collecting modlogs from *subscribed* communities

chris-y committed:
update deployment
currently set to not collect modlogs
Suggest to run this on the modbots, but due to a lemmy bug we need to sub to the correct communities with them
Suggest to enable MODLOG_PM only after it has run at least once, otherwise we risk PMing somebody with an old report

chris-y committed:
Import modlog

chris-y committed:
Bump pythorhead to 0.31.0
Required for modlog

chris-y committed:
Merge branch 'fnic-lemmy:main' into modlog

chris-y committed:
Always set pm_modlogs to a value

chris-y committed:
Merge branch 'fnic-lemmy:main' into modlog

chris-y committed:
typo

chris-y commented:
These are the flags in deploy.sh we will need to change when this is merged and the bot accounts are checked:
```
get_modlog="FALSE" # set to true after we've checked the modbots are subbed to the right communities
modlog_pm="FALSE"  # set to true after one execution
```

M4rtineski commented:
Ye, I saw your issue on that. Personally I don't like needing to run it once before turning on the PMs or risk spamming users with their deleted posts. Every time a new community would be added the process would be needed to be repeated and it would be a high risk low return to me... Being confronted by users about their old deletions or reminding them unnecessarily about it is the last thing I'd want.

chris-y commented:
New communities won't have any reports anyway....

chris-y commented:
I'll think about how I can prevent it for communities which haven't been seen before.

M4rtineski commented:
I meant newly added communities when we get modded and not actually brand new ones. For example if we find an abandoned art community or something similar to raining. I will forget that this setting/problem even exists soon after this would be merged and I don't want to be taken off guard by that later.

chris-y committed:
Add community to an allow list once we've marked the existing logs as processed
This avoids old logs being seen

chris-y commented:
I've fixed this - notifications and PMs only happen now after the community has been scanned once (marking the existing modlogs as read).  I've tested by subbing to a test community which already has modlogs in it, and I didn't get notified of those but I did get notified of a post I removed afterwards.

M4rtineski commented:
Alright. Do you want to merge this now or after the communities were subscribed to?

chris-y commented:
Probably should go around first and ensure the fnicmodbot accounts are only subscribed to the communities they moderate.

chris-y commented:
> Probably should go around first and ensure the fnicmodbot accounts are only subscribed to the communities they moderate.

If you don't get to this tonight I'll go round and check them all tomorrow.

chris-y committed:
Enable modlog scanning
